src/olympia/reviews/models.py

```diff
@@ -152,7 +152,7 @@ def delete(self, user_responsible=None):
         review_was_moderated = False
         # Log deleting reviews to moderation log,
         # except if the author deletes it
-        if not self.addon.has_author(user_responsible):
+        if user_responsible != self.user:
```

Comment:
Right. Test added that covers when the user that deletes the review is the author of the review. The other case is already covered by `test_moderator_delete`.